### PR TITLE
#164688519 Vote for Answers (PATCH)

### PR DESCRIPTION
**Pull request by matthenge**
### What does this PR do?
Add Upvote and Downvote Answers PATCH Endpoints
### What are the tasks to be completed?
Ensure the following are working
```
1. A user can upvote or downvote an answer
2. The data is persisted in the database
3. Ensure a user cannot upvote more than once or downvote more than once for the same answer
4. PATCH /api/meetups/{meetupId}/questions/{questionId}/answers/{answerId}/upvote
5. PATCH /api/meetups/{meetupId}/questions/{questionId}/answers/{answerId}/downvote
```
### Local Development Setup
 - First Create python virtual env
 ```
  $ virtualenv -p python .venv
 ```
 - Install Requirements
 ```
  $ pip install -r requirements.txt
 ```
 - Copy .env-example to .env and set config
 ```
  $ cp .env-example .env
 ```
 - Create postgres database
 ```
  $ sudo su postgres
  $ psql
  postgres=# CREATE USER django WITH PASSWORD 'password';
  postgres=# ALTER ROLE django SET client_encoding TO 'utf8';
  postgres=# ALTER ROLE django SET default_transaction_isolation TO 'read committed';
  postgres=# ALTER ROLE django SET timezone TO 'UTC';
  postgres=# CREATE DATABASE drf;
  postgres=# GRANT ALL PRIVILEGES ON DATABASE drf TO django;
  postgres=# \q
  $ exit
````
### Create admin user
Create admin user using your email and password
```
1. cd api/
2. python manage.py createsuperuser
 ```
 - Run Server
 ```
  $ python manage.py runserver
 ```
a. **After cloning the repo:-**
```
$ git checkout  ft-vote-answer-164688519
$ cd config/
 replace line 18 in swagger.py file with your local host server
```
b. **Test Endpoints**
```
* Log in to be authorized to upvote and downvote answer
* launch swagger documentation using browser  {your local -host}/api/schema/swagger
* Go to Answers collection and test path endpoints for upvote and downvote
```
### Sample Screenshots 
Screenshot from 2019-03-25 19-58-40
Screenshot from 2019-03-25 19-58-14

### What are the relevant pivotal tracker stories?
[#164688519](https://www.pivotaltracker.com/story/show/164688519)

### CODE REVIEW

**Review by Amoswachira — CHANGES_REQUESTED** [review]
i see you have some merger conflicts

**Comment by matthenge** [comment]
> i see you have some merger conflicts

Yeah will work on those ASAP @Amoswachira thanks

File: api/config/swagger.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 def schema_view_swagger(request):
     schema = coreapi.Document(
         title='Questioner API',
-        url='https://questioners-two-staging.herokuapp.com/',  # replace with your url
+        url='http://127.0.0.1:8000/',  # replace with your url
```
Comment (Bakley) [inline api/config/swagger.py]: This is supposed to be the heroku link

Comment (matthenge) [reply api/config/swagger.py]: @Bakley thanks for catching that. will change in the next push

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]
Hello @matthenge ,
Please rebase from develop, resolve merge conflicts and  ensure you have all documentations already in heroku on your branch.
### Here is what heroku has on Questions documentation
Screenshot from 2019-03-21 22-02-00
### Here is what your branch has on Questions
Screenshot from 2019-03-21 22-05-28

File: api/answers/views.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+from .models import Answer, AnswerVote
+from questions.models import Question
+from meetups.models import Meetup
+from .serializers import AnswerSerializer
+
+
+# Create your views here.
+
+
+class AnswersPostView(APIView):
```
Comment (Ogutu-Brian) [inline api/answers/views.py:25]: Hello @matthenge ,
@Bakley  coded this endpoint for post answer. I can see you have created it too...Did you sync with him on this?

Comment (matthenge) [reply api/answers/views.py:25]: yeah I just needed something to work with @Ogutu-Brian

File: api/answers/models.py [hunk 3]
```diff
@@ -25,3 +25,39 @@ class Meta:
...
+                                              vote_type='downvote').count()
+        votes = upvotes - downvotes
+        resultset = {
+            'upvotes': upvotes,
+            'downvotes': downvotes,
+            'votes': votes
+        }
+        return resultset
+
+    def __str__(self):
```
Comment (a-braham) [inline api/answers/models.py:40]: This `__str__(self)` is already defined, There is no need of defining it twice

**Review by Ogutu-Brian — DISMISSED** [review]

File: api/answers/models.py (hunk 3 above)
Comment (matthenge) [reply api/answers/models.py:40]: okay which one are you referring to? @a-braham Please clarify

**Review by IssaIan — DISMISSED** [review]

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]
Hello @matthenge ,
In views.py for questions app, the class `ViewQuestionsView` has been repeated. You can delete the one that starts from line 56.

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]
Hello @matthenge ,
 Avoid the many return statements in your up-vote and down-vote Views. In addition the up-vote and the down-vote views have the same logic, you could define a vote method that can be shared by the two views or one class could inherit from another class to avoid redundancy.

**Comment by matthenge** [comment]
> Hello @matthenge ,
> In views.py for questions app, the class `ViewQuestionsView` has been repeated. You can delete the one that starts from line 56.

Already done @Ogutu-Brian

**Comment by matthenge** [comment]
> Hello @matthenge ,
> Avoid the many return statements in your up-vote and down-vote Views. In addition the up-vote and the down-vote views have the same logic, you could define a vote method that can be shared by the two views or one class could inherit from another class to avoid redundancy.

will work on that. I have been thinking about that too

**Review by Amoswachira — APPROVED** [review]

**Review by Ogutu-Brian — APPROVED** [review]

**Review by Bakley — DISMISSED** [review]
Please increase your test coverage as we can't merge PRs without a green check.

**Pull request merged**